src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,960 @@
+            pixel_values
+        )  # (batch_size, visual_sequence_length, hidden_size)
+        visual_attention_mask = attention_mask.new_ones(
+            visual_embedding_output.shape[:2]
+        )  # (batch_size, visual_sequence_length)
+        pt_mask = torch.ones(attention_mask.shape[0], 10)
+        attention_mask = torch.cat([pt_mask.long(), attention_mask, visual_attention_mask], dim=-1)
+        text_prompt = self.text_prompt.expand(text_embedding_output.shape[0], -1, -1)
+        # (batch_size, sequence_length + visual_sequence_length, hidden_size)
+        embedding_output = torch.cat([text_prompt, text_embedding_output, visual_embedding_output], dim=1)
+
+        extended_attention_mask = self.get_extended_attention_mask(attention_mask, input_shape).to(device)
+        encoder_outputs = self.encoder(
+            embedding_output,
+            attention_mask=extended_attention_mask,
+            head_mask=self.get_head_mask(head_mask, self.config.num_hidden_layers),
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+        last_hidden_state = encoder_outputs.last_hidden_state if return_dict else encoder_outputs[0]
+        pooled_output = self.pooler(last_hidden_state)
+        last_hidden_state = self.dropout(last_hidden_state)
+        pooled_output = self.dropout(pooled_output)
+        if not return_dict:
+            return (
+                last_hidden_state,
+                pooled_output,
+            ) + encoder_outputs[1:]
+
+        return BaseModelOutputWithPooling(
+            last_hidden_state=last_hidden_state,
+            pooler_output=pooled_output,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+
+# Pad single images in the surroundings.
+class TvpPadPrompter(nn.Module):
+    def __init__(self, image_size, prompt_size):
+        super(TvpPadPrompter, self).__init__()
+        pad_size = prompt_size
+        image_size = image_size
+
+        self.base_size = image_size - pad_size * 2
+        self.pad_up = nn.Parameter(torch.randn([1, 1, 3, pad_size, image_size]))
+        self.pad_down = nn.Parameter(torch.randn([1, 1, 3, pad_size, image_size]))
+        self.pad_left = nn.Parameter(torch.randn([1, 1, 3, image_size - pad_size * 2, pad_size]))
+        self.pad_right = nn.Parameter(torch.randn([1, 1, 3, image_size - pad_size * 2, pad_size]))
+
+    def forward(self, x):
+        base = torch.zeros(1, 1, 3, self.base_size, self.base_size)
+        prompt = torch.cat([self.pad_left, base, self.pad_right], dim=4)
+        prompt = torch.cat([self.pad_up, prompt, self.pad_down], dim=3)
+        prompt = torch.cat(x.size(1) * [prompt], dim=1)
+        prompt = torch.cat(x.size(0) * [prompt])
+        return prompt
+
+
+# Pad single images only at the bottom.
+class TvpDownPadPrompter(nn.Module):
```

Comment:
can't the next class be applied on images as well ?

Reply:
unless the checkpoint for the pad_down parameter is different?

Reply:
You were right, we don't need another class for the single image since we can set `num_frames=1`. I will delete these useless classes.